### PR TITLE
DStyle: Constraints on declarations should have the same indentation level

### PR DESCRIPTION
**Pull request by wilzbach**
More unification of our codebase :)

(consequence of https://github.com/dlang/dlang.org/pull/1534)

While this isn't automatically enforced yet, a Dscanner plugin has already been written and is in the [merge queue](https://github.com/Hackerpilot/Dscanner/pull/394) (in fact it was used as a basis to generate these automatic replacements) and getting the Phobos codebase to a common style can be done beforehand.

=> So I am sorry that the diff is quite large - however, the idea is that we do this once & don't have to deal with this problem in the future anymore.

### CODE REVIEW

**Comment by jmdavis** [comment]
Can we _please_ stop adding random formatting rules to the style guide. IMHO, this makes the constraints harder to read.

**Comment by wilzbach** [comment]
> Can we please stop adding random formatting rules to the style guide.

Reason: https://github.com/dlang/phobos/pull/4964#discussion_r93830907

**Comment by dukc** [comment]
Same thoughts vith Davis. Not everything needs, nor should, be standartized. Granted, readability comes first, writablity second. But that does not mean one should sacrifice all writablity for readability.

Note, I am not criticizing the work done here, I am criticizing #4964 comment.

**Comment by JackStouffer** [comment]
Is there a way in which only the patch is scanned? So that new code conforms and the old stuff doesn't necessarily have to be touched.

@jmdavis @andralex Would this be a good compromise?

**Comment by andralex** [comment]
>As this PR shows, there's a lot of code in Phobos which doesn't follow that indentation, and it wasn't a problem before. 

It is a problem. Phobos is a messy mix of styles, which makes it gratuitously difficult to work with. Let's not argue this please - it's subjective so a decision needs be made. @wilzbach please rebase and let's pull. Thanks!

Expect future improvements to the style guide followed by automated enforcements.

**Comment by andralex** [comment]
@JackStouffer one way or another we get to the same point. I'd be more worried if we had no mechanical enforcement (in which case we'd look at doing this once in a while).

File: std/algorithm/comparison.d [hunk 1]
```diff
@@ -1727,7 +1728,7 @@ Returns:
     $(D true) if both ranges have the same length, $(D false) otherwise.
 */
 bool isSameLength(Range1, Range2)(Range1 r1, Range2 r2)
-    if (isInputRange!Range1 &&
+if (isInputRange!Range1 &&
         isInputRange!Range2 &&
         !isInfinite!Range1 &&
         !isInfinite!Range2)
```
Comment (sprinkle131313) [inline std/algorithm/comparison.d]: I feel these following lines should be reduced to be indented once now as well?

**Pull request closed without merging**